Added instructions to integrate the multicluster plugin in coredns (#11)
kubernetes-sigs/sig-multicluster-site: site-src/guides/coredns.md

# Configuring CoreDNS with the multicluster option

> The [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) is necessary to be able to reference multicluster services properly. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this fonctionality.

Reviewer: Minor typo

```suggestion
The [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) is necessary to be able to reference multicluster services properly. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this functionality.
```

> The [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) is necessary to be able to reference multicluster services properly. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this functionality.

Reviewer 2: ```suggestion
There is a ready made [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) you can use as the DNS component against MCS API implementations using EndpointSlices. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this functionality.
```

coredns-565d847f94-zzdrd                          1/1     Running   0          5m20s
```

Before proceding with the re-compilation of CoreDNS, let's retrieve its current configuration by executing the command `coredns -plugins` in one of those pods.

```
kubectl exec -ti coredns-565d847f94-brvlb -n kube-system -- /coredns -plugins
```
**Output (Do Not Copy)**
```
Server types:
  dns

Caddyfile loaders:
  flag
  default

Other plugins:
  dns.acl
  dns.any
  dns.auto
  dns.autopath
  dns.azure
  dns.bind
  dns.bufsize
  dns.cache
  dns.cancel
  dns.chaos
  dns.clouddns
  dns.debug
  dns.dns64
  dns.dnssec
  dns.dnstap
  dns.erratic
  dns.errors
  dns.etcd
  dns.file
  dns.forward
  dns.geoip
  dns.grpc
  dns.header
  dns.health
  dns.hosts
  dns.k8s_external
  dns.kubernetes
  dns.loadbalance
  dns.local
  dns.log
  dns.loop
  dns.metadata
  dns.minimal
  dns.nsid
  dns.pprof
  dns.prometheus
  dns.ready
  dns.reload
  dns.rewrite
  dns.root
  dns.route53
  dns.secondary
  dns.sign
  dns.template
  dns.tls
  dns.trace
  dns.transfer
  dns.whoami
  on
```

This list is important as we will use it and just add the multicluster plugin to recompile CoreDNS.

## Re-compiling CoreDNS

> The easiest way to recompile CoreDNS and generate a new container image is to use the instructions on the [official CoreDNS source repository](https://github.com/coredns/coredns#compilation-from-source). We specifically recommend the compilation with Docker which includes a read-to-use Go environment.

Reviewer: I think you mean ready to use?

```suggestion
The easiest way to recompile CoreDNS and generate a new container image is to use the instructions on the [official CoreDNS source repository](https://github.com/coredns/coredns#compilation-from-source). We specifically recommend the compilation with Docker which includes a ready-to-use Go environment.
```

> Prior to trigger the compilation, it is necessary to modify the `plugin.cfg` file. First, it is highly recommended to keep the same plugins as the ones found in the previous paragraph.

Reviewer: Improve grammar:

```suggestion
Prior to triggering the compilation, it is necessary to modify the `plugin.cfg` file. First, it is highly recommended to keep the same plugins as the ones found in the previous paragraph.
```
